### PR TITLE
Adding Bazel support; dep cleanup; other cleanup

### PR DESCRIPTION
**Pull request by garrettjonesgoogle**
For those not familiar with Bazel:

- BUILD.bazel declares the targets that can be built
- WORKSPACE declares the external "projects" (basically where to find dependencies)

Bazel made me add all directly referenced dependencies, which made it clear we were implicitly using code from several libraries but not declaring them in the build.gradle, so I added them there too. 

I had to do a couple other random things to get gapic-generator to build & run through Bazel.

### CODE REVIEW

File: BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,57 @@
+java_plugin(
+    name = "auto_value_plugin",
+    processor_class = "com.google.auto.value.processor.AutoValueProcessor",
+    deps = ["@com_google_auto_value_auto_value//jar"],
+)
+
+java_library(
+    name = "auto_value",
+    exported_plugins = [":auto_value_plugin"],
+    exports = ["@com_google_auto_value_auto_value//jar"],
```
Comment (vchudnov-g) [inline BUILD.bazel]: Does it make sense to move this `exports` to the `"auto_value_plugin"` target?

(Just learning about `exports` and `java_plugin`--I've not run into them before)

File: WORKSPACE [hunk 2]
```diff
@@ -0,0 +1,142 @@
+workspace(name = "com_google_api_gapic_generator")
```
Comment (vchudnov-g) [inline WORKSPACE]: Would it make sense to add a file comment with the date these were generated?

File: BUILD.bazel (hunk 1 above)
Comment (garrettjonesgoogle) [reply BUILD.bazel]: This was the recommended way to do it... I'm not sure the reason for your suggestion, do you see a problem with it how it is?

File: WORKSPACE (hunk 2 above)
Comment (garrettjonesgoogle) [reply WORKSPACE]: This wasn't generated, it was handwritten. (so I don't think it makes sense...)

File: BUILD.bazel (hunk 1 above)
Comment (garrettjonesgoogle) [reply BUILD.bazel]: I'm going to merge because it works as-is; we can discuss offline if you think something should change later.

**Pull request merged**